pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,136 @@
+	//
+	// +kubebuilder:validation:MaxItems=1
+	// +optional
+	ServiceNetwork []IPNet `json:"serviceNetwork,omitempty"`
+}
+
+// MachineNetworkEntry is a single IP address block for node IP blocks.
+type MachineNetworkEntry struct {
+	// CIDR is the IP block address pool for machines within the cluster.
+	CIDR IPNet `json:"cidr"`
+}
+
+// ClusterNetworkEntry is a single IP address block for pod IP blocks. IP blocks
+// are allocated with size 2^HostSubnetLength.
+type ClusterNetworkEntry struct {
+	// CIDR is the IP block address pool.
+	CIDR IPNet `json:"cidr"`
+
+	// HostPrefix is the prefix size to allocate to each node from the CIDR.
+	// For example, 24 would allocate 2^8=256 adresses to each node. If this
+	// field is not used by the plugin, it can be left unset.
+	// +optional
+	HostPrefix int32 `json:"hostPrefix,omitempty"`
+}
+
+// Proxy defines the proxy settings for the cluster.
+// At least one of HTTPProxy or HTTPSProxy is required.
+type Proxy struct {
+	// HTTPProxy is the URL of the proxy for HTTP requests.
+	// +optional
+	HTTPProxy string `json:"httpProxy,omitempty"`
+
+	// HTTPSProxy is the URL of the proxy for HTTPS requests.
+	// +optional
+	HTTPSProxy string `json:"httpsProxy,omitempty"`
+
+	// NoProxy is a comma-separated list of domains and CIDRs for which the proxy should not be used.
+	// +optional
+	NoProxy string `json:"noProxy,omitempty"`
+}
+
+type ClusterProgressInfo struct {
+	// progress info
+	ProgressInfo string `json:"progressInfo,omitempty"`
+	// Time at which the cluster install progress was last updated.
+	LastProgressUpdateTime *metav1.Time `json:"lastProgressUpdateTime,omitempty"`
+}
+
+type HostNetwork struct {
+	// cidr
+	Cidr string `json:"cidr,omitempty"`
+	// host ids
+	HostIds []string `json:"hostIDs"`
+}
+
+// ClusterStatus defines the observed state of Cluster
+type ClusterStatus struct {
+	State                        string              `json:"state,omitempty"`
+	StateInfo                    string              `json:"stateInfo,omitempty"`
+	HostNetworks                 []HostNetwork       `json:"hostNetworks,omitempty"`
+	InstallationStartTime        string              `json:"installationStartTime,omitempty"`
+	InstallationCompletionTime   *metav1.Time        `json:"installationCompletionTime,omitempty"`
+	Hosts                        int                 `json:"hosts,omitempty"`
+	Progress                     ClusterProgressInfo `json:"progress,omitempty"`
+	ValidationsInfo              string              `json:"validationsInfo,omitempty"`
+	ConnectivityMajorityGroups   string              `json:"connectivityMajorityGroups,omitempty"`
+	LastUpdateTime               *metav1.Time        `json:"lastUpdateTime,omitempty"`
+	ControllerLogsCollectionTime *metav1.Time        `json:"controllerLogsCollectionTime,omitempty"`
+	// Display api errors
+	Error string `json:"error,omitempty"`
+	// Cluster ID
+	// TODO: probably need to work with labels
+	ID string `json:"id,omitempty"`
```

Comment:
not sure what these are for?

Reply:
In the REST API we refer to clusters by UUID, we need some way to map the cluster resource to UUID

Reply:
Is this the same UUID generated by installer for each cluster? If so Hive has this for us in cd.Spec.ClusterMetadata.ClusterID.

Reply:
No, this ID gets created before the installer is ever called.

Reply:
Is it a fully distinct UUID, or is it injected into the installer and the cluster itself for things like telemetry etc?

Can it be restored in the event the value was lost from Status? (i.e. disaster recovery with velero or similar) If not we'll need to move to Spec.

Reply:
It is a distinct UUID.  And you're right, it cannot be recovered.

Reply:
Ok I will move this to Spec somewhere.

Reply:
@filanov  confirms no longer needed assisted is just storing namespace+name key in db to identify.